examples/petset/mongodb-petset.yaml

```diff
@@ -0,0 +1,136 @@
+    displayName: "MongoDB Docker Image"
+    description: "The name of the MongoDB docker image"
+    value: "openshift/mongodb-24-centos7"
+    required: true
+
+  - name: OPENSHIFT_SERVICE_NAME
+    displayName: "OpenShift Service Name"
+    description: "The name of the OpenShift Service exposed for the database"
+    value: mongodb
+    required: true
+
+objects:
+  # A headless service to create DNS records
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      name: "${OPENSHIFT_SERVICE_NAME}"
+    spec:
+      # for a headless service proxying is not required
+      clusterIP: None
+      # the list of ports that are exposed by this service
+      ports:
+        - name: mongodb
+          port: 27017
+      # will route traffic to pods having labels matching this selector
+      selector:
+        node-type: "${OPENSHIFT_SERVICE_NAME}-node"
+  - apiVersion: apps/v1alpha1
+    kind: PetSet
+    metadata:
+      name: mongodb
+    spec:
+      # pets get DNS/hostnames that follow the pattern: ${metadata.name}-NUM.${spec.serviceName}.default.svc.cluster.local
+      serviceName: "${OPENSHIFT_SERVICE_NAME}"
+      replicas: 3
+      # describes the pod that will be created if insufficient replicas are detected
+      template:
+        metadata:
+          # this label will be used for count running pods
+          labels:
+            node-type: "${OPENSHIFT_SERVICE_NAME}-node"
+          annotations:
+            # 'false' pauses a petset after creation of each pet, to avoid it we set it to 'true'
+            pod.alpha.kubernetes.io/initialized: "true"
+        spec:
+          containers:
+            - name: mongo-container
+              image: "${OPENSHIFT_MONGODB_IMAGE}"
+              # list of ports to expose from the container
+              ports:
+                - containerPort: 27017
+              args:
+                - "run-mongod-pet"
+              volumeMounts:
+                - name: mongo-data
+                  mountPath: "/var/lib/mongodb"
+              redinessProbe:
+                tcpSocket:
+                  port: 27017
+              env:
+                - name: MONGODB_USER
+                  value: "${MONGODB_USER}"
+                - name: MONGODB_PASSWORD
+                  value: "${MONGODB_PASSWORD}"
+                - name: MONGODB_DATABASE
+                  value: "${MONGODB_DATABASE}"
+                - name: MONGODB_ADMIN_PASSWORD
+                  value: "${MONGODB_ADMIN_PASSWORD}"
+                - name: MONGODB_REPLICA_NAME
+                  value: "${MONGODB_REPLICA_NAME}"
+                - name: MONGODB_KEYFILE_VALUE
+                  value: "${MONGODB_KEYFILE_VALUE}"
+          securityContext:
+            # in case of using hostPath/nfs: this should be the group id of the
+            # directory that will be used as PVs. To allow access of mongodb
+            # to the directory and files, we're adding this GID to the process.
+            supplementalGroups: [ 1000 ]
```

Comment:
@omron93 This option adds additional group to the user that is executing mongodb within the pod. By doing this, mongodb has permissions to modify fils on persistent volume (that is a directory on the host in our example).

See also: https://docs.openshift.com/enterprise/3.2/install_config/persistent_storage/pod_security_context.html#supplemental-groups

>Why not to use 184 - mongodb reserved uid/gid?

Because on my host /tmp/pv_* directories owned by user with group 1000